server/config/middleware.js

```diff
@@ -132,69 +130,88 @@ module.exports = function (app, express) {
   var client = new FitbitClient('22B2V3', '1fb7088fd54576f1025f23a88d03f371');
   var redirect_uri = 'http://localhost:8000/auth/fitbit/callback';
   var scope =  [ 'activity' ];
-  console.log('client =', client);
-  app.get('/auth/fitbit', function(req, res, next) {
-        
+
+  app.get('/auth/fitbit', 
+    function(req, res, next) {
       var authorization_uri = client.getAuthorizationUrl(redirect_uri, scope);
-      
       res.redirect(authorization_uri);
   }); 
         
-  app.get('/auth/fitbit/callback', function(req, res, next) {
-  
-      var code = req.query.code;
-      // console.log('code =', code);
-      
-      client.getToken(code, redirect_uri)
-          .then(function(token) {
-            token = token.token;
-            //write access token and refresh token to database under this user
-
-              // ... save your token on db or session... 
-              // console.log('token =', token);
-              // then redirect
-              res.end();
-              // res.redirect(302, '/user');
-  
-          })
-          .catch(function(err) {
-              console.log('error');
-              // something went wrong.
-              res.send(500, err);
-          
+  app.get('/auth/fitbit/callback', ensureAuthenticated, function(req, res, next) {
+    var code = req.query.code;
+    client.getToken(code, redirect_uri)
+    .then(function(token) {
+      token = token.token;
+      db.User.findOne({ where: {fbID: req.user.dataValues.fbID} })
+      .then(function(user) {
+          db.AccountFitBit.findOrCreate({ 
+            where: { 
+              fitBitID: token.user_id} })
+          .then(function(account) {
+            db.AccountFitBit.findOne({fitBitID: token.user_id})
+            .then(function(account) {
+              account.update({
+                fitBitAccessToken: token.access_token, 
+                fitBitRefreshToken: token.refresh_token})
+              .then(function(accountObj) {
+                client.getTimeSeries({
+                  access_token: token.access_token,
+                  refresh_token: token.refresh_token})
+                .then(function(results) {
```

Comment:
Consider refactoring this to do a single update step to make the code cleaner/more compact